### ISSUE TITLE
pkgdown failing

### ISSUE THREAD

**Issue by hulsiejames** [udsleeds/openinfra#84]
pkgdown is failing in [actions](https://github.com/udsleeds/openinfra/runs/7425801909?check_suite_focus=true). 

Main message: 
```
-- Building function reference -------------------------------------------------
Writing 'reference/index.html'
Error in library(pkg$package, character.only = TRUE) : 
  there is no package called ‘openinfra’
Calls: <Anonymous> ... build_site_local -> build_reference -> examples_env -> library
```

devtools::check() shows nothing breaking: 
```
── R CMD check results ───────────────────────────────────────────────────────────────────────────────────────────────────────────────── openinfra 0.0.0.9000 ────
Duration: 4m 41.5s

❯ checking for portable file names ... WARNING
  Found the following file with a non-portable file name:
    vignettes/rsconnect/documents/data_packs.Rmd/rpubs.com/rpubs/Publish Document.dcf
  These are not fully portable file names.
  See section ‘Package structure’ in the ‘Writing R Extensions’ manual.

❯ checking installed package size ... NOTE
    installed size is 29.1Mb
    sub-directories of 1Mb or more:
      data   2.5Mb
      doc   26.4Mb

❯ checking top-level files ... NOTE
  Non-standard files/directories found at top level:
    ‘5km_Leeds_OSM_highways.html’ ‘Python_scripts’ ‘Quarto_docs’
    ‘ResCompLeedsCon’ ‘lida_report’ ‘modelling_world_2022’ ‘pkgdown’
    ‘presentations’ ‘references.bib’ ‘setup.R’ ‘sotm2022’

❯ checking dependencies in R code ... NOTE
  Namespaces in Imports field not imported from:
    ‘od’ ‘osmextract’ ‘pct’ ‘sf’ ‘stats19’ ‘tmap’
    All declared Imports should be used.

0 errors ✔ | 1 warning ✖ | 3 notes ✖
```

devtools::session_info() seems more insightful: 
```
─ Session info ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.1 (2022-06-23)
 os       Ubuntu 20.04.4 LTS
 system   x86_64, linux-gnu
 ui       RStudio
 language en_GB:en
 collate  en_GB.UTF-8
 ctype    en_GB.UTF-8
 tz       Europe/London
 date     2022-07-20
 rstudio  2022.02.3+492 Prairie Trillium (desktop)
 pandoc   2.17.1.1 @ /usr/lib/rstudio/bin/quarto/bin/ (via rmarkdown)

─ Packages ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 ! package      * version    date (UTC) lib source
   abind          1.4-5      2016-07-21 [1] CRAN (R 4.2.0)
   assertthat     0.2.1      2019-03-21 [3] CRAN (R 4.0.0)
   base64enc      0.1-3      2015-07-28 [3] CRAN (R 4.0.0)
   brio           1.1.3      2021-11-30 [3] CRAN (R 4.1.2)
   cachem         1.0.6      2021-08-19 [1] CRAN (R 4.2.0)
   callr          3.7.0      2021-04-20 [3] CRAN (R 4.0.5)
   class          7.3-20     2022-01-13 [4] CRAN (R 4.1.2)
   classInt       0.4-7      2022-06-10 [1] CRAN (R 4.2.0)
   cli            3.3.0      2022-04-25 [3] CRAN (R 4.2.0)
   codetools      0.2-18     2020-11-04 [4] CRAN (R 4.0.3)
   commonmark     1.8.0      2022-03-09 [1] CRAN (R 4.2.0)
   crayon         1.5.1      2022-03-26 [3] CRAN (R 4.1.3)
   crosstalk      1.2.0      2021-11-04 [1] CRAN (R 4.2.0)
   curl           4.3.2      2021-06-23 [3] CRAN (R 4.1.0)
   DBI            1.1.3      2022-06-18 [1] CRAN (R 4.2.0)
   desc           1.4.1      2022-03-06 [3] CRAN (R 4.1.3)
   devtools       2.4.3      2021-11-30 [1] CRAN (R 4.2.1)
   dichromat      2.0-0.1    2022-05-02 [1] CRAN (R 4.2.0)
   digest         0.6.29     2021-12-01 [3] CRAN (R 4.1.2)
   dplyr          1.0.9      2022-04-28 [3] CRAN (R 4.2.0)
   e1071          1.7-11     2022-06-07 [1] CRAN (R 4.2.0)
   ellipsis       0.3.2      2021-04-29 [3] CRAN (R 4.0.5)
   evaluate       0.15       2022-02-18 [3] CRAN (R 4.1.2)
   fansi          1.0.3      2022-03-24 [3] CRAN (R 4.1.3)
   fastmap        1.1.0      2021-01-25 [3] CRAN (R 4.0.3)
   fs             1.5.2      2021-12-08 [3] CRAN (R 4.1.2)
   generics       0.1.3      2022-07-05 [3] CRAN (R 4.2.1)
   glue           1.6.2      2022-02-24 [3] CRAN (R 4.1.2)
   hms            1.1.1      2021-09-26 [3] CRAN (R 4.1.1)
   htmltools      0.5.3      2022-07-18 [1] CRAN (R 4.2.1)
   htmlwidgets    1.5.4      2021-09-08 [3] CRAN (R 4.1.1)
   KernSmooth     2.23-20    2021-05-03 [4] CRAN (R 4.0.5)
   knitr          1.39       2022-04-26 [1] CRAN (R 4.2.1)
   lattice        0.20-45    2021-09-22 [4] CRAN (R 4.2.0)
   leafem         0.2.0      2022-04-16 [1] CRAN (R 4.2.0)
   leaflet        2.1.1      2022-03-23 [1] CRAN (R 4.2.0)
   leafsync       0.1.0      2019-03-05 [1] CRAN (R 4.2.0)
   lifecycle      1.0.1      2021-09-24 [3] CRAN (R 4.1.1)
   lwgeom         0.2-8      2021-10-06 [1] CRAN (R 4.2.0)
   magrittr       2.0.3      2022-03-30 [3] CRAN (R 4.1.3)
   memoise        2.0.1      2021-11-26 [1] CRAN (R 4.2.0)
   od             0.4.0      2022-04-18 [1] CRAN (R 4.2.0)
 P openinfra    * 0.0.0.9000 2022-07-18 [?] Github (udsleeds/openinfra@25fcd92)
   osmextract     0.4.0.9000 2022-06-22 [1] Github (ropensci/osmextract@f1cccc7)
   pct            0.9.3      2021-11-02 [1] CRAN (R 4.2.1)
   pillar         1.8.0      2022-07-18 [1] CRAN (R 4.2.1)
   pkgbuild       1.3.1      2021-12-20 [1] CRAN (R 4.2.1)
   pkgconfig      2.0.3      2019-09-22 [3] CRAN (R 4.0.0)
   pkgload        1.3.0      2022-06-27 [1] CRAN (R 4.2.1)
   png            0.1-7      2013-12-03 [1] CRAN (R 4.2.0)
   prettyunits    1.1.1      2020-01-24 [3] CRAN (R 4.0.0)
   processx       3.7.0      2022-07-07 [3] CRAN (R 4.2.1)
   proxy          0.4-27     2022-06-09 [1] CRAN (R 4.2.0)
   ps             1.7.1      2022-06-18 [3] CRAN (R 4.2.1)
   purrr          0.3.4      2020-04-17 [3] CRAN (R 4.0.0)
   R6             2.5.1      2021-08-19 [3] CRAN (R 4.1.1)
   raster         3.5-21     2022-06-27 [1] CRAN (R 4.2.1)
   rcmdcheck      1.4.0      2021-09-27 [1] CRAN (R 4.2.1)
   RColorBrewer   1.1-3      2022-04-03 [3] CRAN (R 4.1.3)
   Rcpp           1.0.9      2022-07-08 [1] CRAN (R 4.2.1)
   readr          2.1.2      2022-01-30 [3] CRAN (R 4.1.2)
   remotes        2.4.2      2021-11-30 [1] CRAN (R 4.2.0)
   rlang          1.0.4      2022-07-12 [1] CRAN (R 4.2.1)
   rmarkdown      2.14       2022-04-25 [3] CRAN (R 4.2.0)
   roxygen2       7.2.0      2022-05-13 [1] CRAN (R 4.2.1)
   rprojroot      2.0.3      2022-04-02 [3] CRAN (R 4.1.3)
   rstudioapi     0.13       2020-11-12 [3] CRAN (R 4.0.3)
   sessioninfo    1.2.2      2021-12-06 [1] CRAN (R 4.2.1)
   sf             1.0-8      2022-07-14 [1] CRAN (R 4.2.1)
   sp             1.5-0      2022-06-05 [1] CRAN (R 4.2.0)
   stars          0.5-5      2021-12-19 [1] CRAN (R 4.2.0)
   stats19        2.0.0      2021-10-29 [1] CRAN (R 4.2.1)
   stringi        1.7.8      2022-07-11 [1] CRAN (R 4.2.1)
   stringr        1.4.0      2019-02-10 [3] CRAN (R 4.0.0)
   terra          1.5-34     2022-06-09 [1] CRAN (R 4.2.0)
   testthat       3.1.4      2022-04-26 [3] CRAN (R 4.2.0)
   tibble         3.1.7      2022-05-03 [3] CRAN (R 4.2.0)
   tidyselect     1.1.2      2022-02-21 [3] CRAN (R 4.1.2)
   tmap           3.3-3      2022-03-02 [1] CRAN (R 4.2.0)
   tmaptools      3.1-1      2021-01-19 [1] CRAN (R 4.2.0)
   tzdb           0.3.0      2022-03-28 [3] CRAN (R 4.1.3)
   units          0.8-0      2022-02-05 [1] CRAN (R 4.2.0)
   usethis        2.1.6      2022-05-25 [1] CRAN (R 4.2.0)
   utf8           1.2.2      2021-07-24 [3] CRAN (R 4.1.0)
   vctrs          0.4.1      2022-04-13 [3] CRAN (R 4.1.3)
   viridisLite    0.4.0      2021-04-13 [3] CRAN (R 4.0.5)
   withr          2.5.0      2022-03-03 [3] CRAN (R 4.1.3)
   xfun           0.31.2     2022-06-29 [1] https://yihui.r-universe.dev (R 4.2.1)
   XML            3.99-0.10  2022-06-09 [3] CRAN (R 4.2.0)
   xml2           1.3.3      2021-11-30 [3] CRAN (R 4.1.2)
   xopen          1.0.0      2018-09-17 [1] CRAN (R 4.2.1)
   yaml           2.3.5      2022-02-21 [3] CRAN (R 4.1.2)

 [1] /home/james/R/x86_64-pc-linux-gnu-library/4.2
 [2] /usr/local/lib/R/site-library
 [3] /usr/lib/R/site-library
 [4] /usr/lib/R/library

 P ── Loaded and on-disk path mismatch.
```

Specifically from session_info(): 
```
!  package      * version    date (UTC)    lib source
P openinfra    * 0.0.0.9000 2022-07-18 [?] Github (udsleeds/openinfra@25fcd92)
P ── Loaded and on-disk path mismatch.
```

Looking into this now - not a major issue, but the website (built by pkgdown) is not updating.

**Comment by Robinlovelace** [comment]
You have accidentally pushed this file I think @hulsiejames 

```
    vignettes/rsconnect/documents/data_packs.Rmd/rpubs.com/rpubs/Publish Document.dcf
```

**Comment by hulsiejames** [comment]
That only appears in devtools::check() as is stored locally within the openinfra directory, created when I hosted an .Rmd to rpubs.
 
The file hasn't been pushed to the repo [vignettes directory](https://github.com/udsleeds/openinfra/tree/main/vignettes) (hopefully), as is still an untracked local file: 

![](https://i.imgur.com/Tn4NQ9v.png)

**Comment by Robinlovelace** [comment]
Ah yes, my bad, thought it was a message from actions but now see it was a local check message.

**Comment by hulsiejames** [comment]
This may have been due to an outdated `pkgdown.yaml` - much like our issue with R-CMD check a few weeks back. 

See PR #85 - which seems to have fixed this (at least on that branch)

**Comment by hulsiejames** [comment]
Fixed with: [b76f28d](https://github.com/udsleeds/openinfra/commit/b76f28dd670dbb3e8ae390f1bd3ec2324218813c)

Closing for now!
